src/nodes.coffee

```diff
@@ -1762,31 +1786,36 @@ exports.JSXElement = class JSXElement extends Base
       closingElement = Object.assign {
         type: 'JSXClosingElement'
         name: Object.assign(
-          @tagName.unwrap().ast(o),
+          tagNameAst(),
           jisonLocationDataToAstLocationData @tagName.base.closingTagNameLocationData
         )
       }, @closingElementLocationData
-      if closingElement.name.type is 'JSXMemberExpression'
+      if closingElement.name.type in ['JSXMemberExpression', 'JSXNamespacedName']
         rangeDiff = closingElement.range[0] - openingElement.range[0] + '/'.length
-        shiftAstLocationData = (node) ->
+        columnDiff = closingElement.loc.start.column - openingElement.loc.start.column + '/'.length
+        shiftAstLocationData = (node) =>
           node.range = [
             node.range[0] + rangeDiff
             node.range[1] + rangeDiff
           ]
           node.start += rangeDiff
           node.end += rangeDiff
           node.loc.start =
-            line: node.loc.start.line
-            column: node.loc.start.column + rangeDiff
+            line: @closingElementLocationData.loc.start.line
```

Comment:
There was a bug in this logic for generating adjusted closing element location data, it wasn't calculating closing element location data correctly when the closing element was on a different line than the opening element

So fixed and added AST location data tests for both `JSXMemberExpression` (`<a.b>`) and `JSXNamespacedName` (`<a:b>`) when the closing tag is on a different line than the opening tag